components/Images.tsx

```diff
@@ -1,31 +1,58 @@
-import React, { useContext, useEffect } from 'react';
+import React, { useContext } from 'react';
 import { Keycapset } from 'typings';
 import Context from '../context';
 import ImageCard from './ImageCard';
+import { getDayDifference } from './StatusLabel'
 
 import Tabs from './Tabs';
+import moment from 'moment';
 
 interface ImagesProps {}
 
 function Images(props: ImagesProps): JSX.Element {
     const {} = props;
     const {
+        filters: {
+            activeTab,
+            availabilityFilter,
+        },
         keycapsets,
-        activeTab,
         searchQuery
     } = useContext(Context);
 
+    const filterByAvailability = (set: Keycapset): boolean => {
+        if(availabilityFilter === 'none') {
+            return true
+        }
+        const {groupbuyEndDate, groupbuyStartDate, isInterestCheck} = set;
+
+        switch (availabilityFilter){
+        case 'ic':
+            return isInterestCheck;
+        case 'waiting':
+            return moment().diff(groupbuyStartDate, 'days') < 0 && !isInterestCheck;
+        case 'gb':
+            return getDayDifference(groupbuyEndDate) > 0 && !isInterestCheck;
+        case 'ended':
+            return !isInterestCheck 
+                && !(getDayDifference(groupbuyEndDate) > 0) 
+                &&  !(moment().diff(groupbuyStartDate, 'days') < 0) 
+        default:
+            return false
```

Comment:
I'm not sure if i like non 'CONStanized' switch cases. Maybe it's nice to add these filters to the `constants.tsx` file?

Reply:
We could also add an Status enum to the typings?